Restructure Documents folders, separate design from compliance requirements, etc. (#45)
Azure/DASH: documentation/README.md

| Service Tunnel & Private Link Service                          | [Parent Folder](stpl-service/README.md) \| [ Design](stpl-service/design/README.md) \| [Compliance Requirements](stpl-service/requirements/README.md)|
| VNET Peering Service                          | [Parent Folder](vnet-peering-service/README.md) \| [ Design](vnet-peering-service/design/README.md) \| [Compliance Requirements](vnet-peering-service/requirements/README.md)|
| Express Route (ER) Service                    |  [Parent Folder](express-route-service/README.md) \| [ Design](express-route-service/design/README.md) \| [Compliance Requirements](express-route-service/requirements/README.md)|
| Encryption Gateway Service                    |  [Parent Folder](encrypt-gw-service/README.md) \| [ Design](encrypt-gw-service/design/README.md) \| [Compliance Requirements](encrypt-gw-service/requirements/README.md)|

# Relationships and Flow of Documents
The diagram below shows how High-Level Descriptions beget Compliance requirements, compliance requirements beget test cases, and test cases are executed by test scripts to produce Test Results.

![dash-specs-flow](images/general/dash-specs-flow.svg).

> 

Reviewer: I am wondering if we can make this a clickable image, where each link point to the related article(s).

Author: This is just a general flow, there are no specific articles per se to click on. Eventually there will be many artifacts at each "level." There could be hundreds of test cases for example.

> Some of the guiding principles for this aproach are:

Reviewer: ```suggestion
The following are some of the guiding principles for this approach:
```

Author: I don't agree with your proposed change, I wrote as if one is reading the final document, not to describe "what I am changing in the old document." That only has historical passing value.

> * Define the objectives and design or proposal without clutter from performance and requirement details.

Reviewer: ```suggestion
* Define the objectives and the design or proposal separately from performance and requirement details.
```

> * Express hard requirements separately from the design and architecture descriptions so that the requirements are easy to define, maintain and reference from other downstream "consumers," e.g. **test cases**. All requirements are identified with some designator which allows traceability in test cases, scripts and results.

Reviewer: ```suggestion
* Describe **hard requirements** separately from the design and architecture descriptions This allows the requirements to be easily defined, maintained, and referenced from other downstream "consumers," e.g., **test cases**. All requirements must be identified with some designator which allows traceability in test cases, scripts and results.
```

> * Strive for simultaneous human- and machine-readable data which can drive test cases. Avoid burying test parameters into the test script so that requirements can be maintained/defined independently from the (often complex) code which executes tests. Many projects exist where only a programmer can locate and ferret out actual test criteria, often expressed as hard-coded constants buried within thousands of lines of test automation code. In contrast, these criteria should be easily accessible, reviewable and maintainable, to anyone familiar with the project.

Reviewer: ```suggestion
* We encourage the creation of simultaneous **human** and **machine-readable** data which can **drive test cases**.  
* We must avoid burying test parameters into the test scripts. This allows the requirements to be maintained/defined independently from the (often complex) code which executes tests. 
* Many projects exist where only a programmer can locate and ferret out actual test criteria, often expressed as hard-coded constants buried within thousands of lines of test automation code. For quality control, these criteria must be easily accessible, reviewable and maintainable, to anyone familiar with the project.
```

> * Complete auditability and tracebility of tests cases, test results, associated specs and DUT/SUT configuration. This means  a test run will record versions of every item including Git Repo commit SHA ids, branches, tags, SW versions, API versions, etc.

Reviewer: ```suggestion
* We advocate complete auditability and traceability of tests cases, test results, associated specs and DUT/SUT configuration. This means a test run will record versions of every item including GitHub Repo commit SHA ids, branches, tags, SW versions, API versions, etc.
```

> * Clear, concise human-readable reports, plus machine-readable results allowing dashboards, rolling-up of results, etc.

Reviewer: ```suggestion
* Clear, concise and to the point human-readable reports, plus machine-readable results allowing dashboards, rolling-up of results, etc.
```
